### PR TITLE
Fix bug 1214670: Sync per-repo

### PR DESCRIPTION
**Pull request by Osmose**
This is an in-progress PR of changing the sync process to sync translations per-repo instead of syncing all repos/locales at once. This allows us to group database updates such that, if a commit to a repo fails, sync will abort and revert all the changes made related to that repo until the next sync. However, other repos will still get their updates if they do not encounter any conflicts.

I'm not happy with the API and want to reorganize this eventually, but shipping it is more important at this point. I'll be working on making the tests pass with these new changes, but in the meantime this is ready for a preliminary review.

@mathjazz r?

### CODE REVIEW

File: pontoon/sync/core.py [hunk 1]
```diff
@@ -2,126 +2,47 @@
...
+        if vcs_entity is None:
+            if db_entity is None:
+                # This should never happen. What? Hard abort.
+                raise ValueError('No entities found for key {0}'.format(key))
             else:
-                # If Pontoon has nothing or has not changed, and the VCS
-                # still has the entity, update Pontoon with whatever may
-                # have changed.
-                changeset.update_db_entity(locale.code, db_entity, vcs_entity)
+                # VCS no longer has the entity, remove it from Pontoon.
```
Comment (mathjazz) [inline pontoon/sync/core.py]: This is a good opportunity to update this comment: obsolete, not remove.

File: pontoon/sync/core.py [hunk 2]
```diff
@@ -191,36 +127,23 @@ def pull_changes(db_project):
...
-                project=db_project,
-                locale=locale,
-                reason=result['message']
-            ))
+    # Use the top translator for this batch as commit author, or
+    # the fake Pontoon user if there are no authors.
+    if len(authors) > 0:
+        commit_author = Counter(authors).most_common(1)[0][0]
+    else:
+        commit_author = User(first_name="Pontoon", email="pontoon@mozilla.com")
```
Comment (mathjazz) [inline pontoon/sync/core.py]: Let's change the first_name to "Mozilla Pontoon" for consistency with vcs.py.

File: pontoon/sync/core.py [hunk 3]
```diff
@@ -191,36 +127,23 @@ def pull_changes(db_project):
...
+
+    commit_message = render_to_string('commit_message.jinja', {
```
Comment (mathjazz) [inline pontoon/sync/core.py]: We should move this file from administration to sync.

**Comment by mathjazz** [comment]
I like what you did here and the code looks great. I'll test it tomorrow.

**Comment by Osmose** [comment]
Tests are fixed, changes made, this is ready for real world testing and merging. :D

**Comment by Osmose** [comment]
So Travis isn't pulling the right commit for some reason, but trust me, the tests are passing™.

**Comment by mathjazz** [comment]
Seems to be working fine on stage with procfile change!

r+

I'd be happy as a little doll if you could print something when each task completes, like "Synced sources for project X" or "Synced translations for project Y".

**Comment by Osmose** [comment]
Added a message at the end of each task as requested. :D

**Pull request merged**